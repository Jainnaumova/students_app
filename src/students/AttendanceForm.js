import React, { Component } from 'react';

import './students.css';

export default class AttendanceForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      attendanceValue: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({ attendanceValue: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.handleChangeAttendanceValue(this.state.attendanceValue);
  }

  render() {
    return (
      <div className='input-container'>
        <form onSubmit={this.handleSubmit}>
          <input
            type='number'
            className='inputForm'
            max='100.00'
            step='any'
            min='0.00'
            value={this.state.attendanceValue}
            onChange={this.handleChange}
          />
          <button type='submit' className='button'>Apply</button>
        </form>
      </div>
    );
  }
}
