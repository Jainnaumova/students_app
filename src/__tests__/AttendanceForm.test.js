import React from 'react';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-15';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import AttendanceForm from '../students/AttendanceForm';

Enzyme.configure({ adapter: new Adapter() });

const component = () => shallow(<AttendanceForm {...props} />);

const props = {
  handleChangeAttendanceValue: jest.fn()
};

describe('AttendanceForm', () => {

  it('should match the shapshot', () => {
    const tree = renderer.create(
      <AttendanceForm { ...props }/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should have correct initial state', () => {
    const expectedState = {
      attendanceValue: ''
    };
    const wrapper = component();
    expect(wrapper.instance().state).toEqual(expectedState);
  });

  it('should update state on input change', () => {
    const event = {preventDefault: () => {}, target: { value: 'abc'} };
    const expectedState = {
      attendanceValue: 'abc'
    };
    const wrapper = component();
    wrapper.instance().handleChange(event);
    expect(wrapper.instance().state).toEqual(expectedState);
  });

  it('should call handleChangeAttendanceValue from props with correct argument', () => {
    const event = {preventDefault: () => {}};
    const wrapper = component();
    wrapper.instance().setState({ attendanceValue: 'xyz' });
    wrapper.instance().handleSubmit(event);
    expect(props.handleChangeAttendanceValue).toHaveBeenCalledWith('xyz');
    expect(props.handleChangeAttendanceValue).toHaveBeenCalledTimes(1);
  });

  it('should call handleSubmit on form submission', () => {
    const handleSubmit = jest.spyOn(AttendanceForm.prototype, 'handleSubmit');
    const event = {preventDefault: () => {}};
    const wrapper = component();
    wrapper.find('form').simulate('submit', event);
    expect(handleSubmit.mock.calls.length).toEqual(1);
    expect(handleSubmit.mock.calls[0][0]).toEqual(event);
  });

});
