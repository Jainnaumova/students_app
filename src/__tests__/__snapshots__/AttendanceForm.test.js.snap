// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttendanceForm should match the shapshot 1`] = `
<div
  className="input-container"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="inputForm"
      max="100.00"
      min="0.00"
      onChange={[Function]}
      step="any"
      type="number"
      value=""
    />
    <button
      className="button"
      type="submit"
    >
      Apply
    </button>
  </form>
</div>
`;
